server/resourcemanager/resource_manager.go

```diff
@@ -32,55 +32,109 @@ type Resource[T ResourceSpec] struct {
 	Spec T      `mapstructure:"spec"`
 }
 
-type SortableHandler interface {
-	SortingFields() []string
-}
-
-type ResourceHandler[T ResourceSpec] interface {
-	SortableHandler
-	SetID(T, id.ID) T
-	List(_ context.Context, take, skip int, query, sortBy, sortDirection string) ([]T, error)
-	Count(_ context.Context, query string) (int, error)
-	Create(context.Context, T) (T, error)
-	Update(context.Context, T) (T, error)
-	Get(context.Context, id.ID) (T, error)
-	Delete(context.Context, id.ID) error
+type Manager interface {
```

Comment:
this interface is mostly useful for passing managers in tests without needing to know the generic concrete type